specification/marketplacecatalog/resource-manager/Solutions/preview/2023-03-01-preview/solutions.json

```diff
@@ -0,0 +1,501 @@
+      "in": "path",
+      "name": "productId",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "required": true,
+      "description": "The product id to update or delete"
+    }
+  },
+  "definitions": {
+    "CreateOrUpdateSolutionRequestPayload": {
+      "type": "object",
+      "properties": {
+        "solutionName": {
+          "type": "string"
+        },
+        "productIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "AddProductToUserSolutionRequest": {
+      "type": "object",
+      "properties": {
+        "productId": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "Product": {
+      "type": "object",
+      "properties": {
+        "uniqueProductId": {
+          "type": "string"
+        },
+        "displayName": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "summary": {
+          "type": "string"
+        },
+        "smallIconUri": {
+          "type": "string"
+        },
+        "publisherDisplayName": {
+          "type": "string"
+        },
+        "productType": {
+          "type": "string"
+        },
+        "pricingTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "storeFronts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ratingAverage": {
+          "type": "number",
+          "format": "double"
+        }
+      },
+      "additionalProperties": false
+    },
+    "RenameUserSolutionRequest": {
+      "description": "The service properties for a given service.",
+      "type": "object",
+      "properties": {
+        "newName": {
+          "type": "string"
+        }
+      }
+    },
+    "Solution": {
+      "description": "The Solution data structure",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SolutionProperties",
+          "description": "The Solution data structure",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SolutionProperties": {
+      "description": "Describes the json payload for solution operations",
+      "type": "object",
+      "properties": {
+        "solutionId": {
+          "type": "string"
+        },
+        "solutionName": {
+          "type": "string"
+        },
+        "products": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Product"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "UserSolutions": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
```

Comment:
collection GET responses must only have "value" and "nextLink" properties

Reply:
done